### PR TITLE
Updated notebook documentation and refactored some code

### LINKED ISSUE #396: [BUG] Clean up notebooks
The Jupyter notebooks were made over a year ago, and a lot has changed in functionality and syntax. This ticket is to scrub them and bring them up to the current version.

### PR DESCRIPTION
**Pull request by cliffburdick**
Closes #396

### CODE REVIEW

**Comment by cliffburdick** [comment]
Closes #396

File: docs_input/notebooks/01_introduction.ipynb [hunk 1]
```diff
@@ -282,16 +298,17 @@
...
+    "Using this form of `slice` requires a template argument with the rank of the new slice. The second parameter to `slice` takes the starting index for each dimension, while the second takes the ending index. To include all values from the beginning on, a special sentinel of `matxEnd` can be used. Similarly, `matxDropDim` is used to indicate this dimension is the one being sliced (i.e. removed). In this case we are slicing the second column of the tensor and all rows, which produces a new 1D tensor containing only the second column of the original tensor. This is equivalent to `t2[:,1]` in Python. \n",
```
Comment (tylera-nvidia) [inline docs_input/notebooks/01_introduction.ipynb:311]: second -> third here, since we added a param

**Comment by tylera-nvidia** [comment]
In several of the examples we have commented out code, do we want to keep that around or should we uncomment it? I don't quiet understand why we comment some of the test code out, and leave others in.

**Comment by cliffburdick** [comment]
> In several of the examples we have commented out code, do we want to keep that around or should we uncomment it? I don't quiet understand why we comment some of the test code out, and leave others in.

it's commented out because it's part of an example they are supposed to fill out for the training

**Review by tylera-nvidia — APPROVED** [review]

**Pull request merged**